tests/pos/notNull.scala

```diff
@@ -1,33 +1,22 @@
 trait Null extends Any
 object Test with
-  def notNull(x: Any): x.type & NotNull =
+  def notNull[A](x: A | Null): x.type & A =
     assert(x != null)
     x.asInstanceOf  // TODO: drop the .asInstanceOf when explicit nulls are implemented
 
   locally {
     val x: (Int | Null) = ???
     val y = x; val _: Int | Null = y
   }
-  locally {
-    val x: (Int | Null) & NotNull = ???
-    val y = identity(x); val yc: Int = y
-    val z = x; val zc: Int = z
-  }
   locally {
     val x: Int | Null = ???
     val y = notNull(identity(x)); val yc: Int = y
     val z = notNull(x); val zc: Int = z
   }
-  locally {
-    val x: Int | Null = ???
-    val y = identity(x).$nn; val yc: Int = y
-    val z = x.$nn; val zc: Int = z
-  }
   class C { type T }
   locally {
     val x: C { type T = Int } = new C { type T = Int }
-    val y: x.$nn.T = 33
+    val xnn: x.type & C { type T = Int } = notNull(x)
```

Comment:
The problem is that the compiler will insert `$nn` on a path, and that the path can appear in a type or a term. I don't see how that could work with the revised version of `notNull`.

Reply:
As I explained at https://github.com/lampepfl/dotty/pull/6344#issuecomment-553431009, I don't think the use in paths is a real issue. We can get away without supporting it.

Reply:
I put my answer on that thread as well, since it covered several earlier remarks. https://github.com/lampepfl/dotty/pull/6344#issuecomment-554062532.

With 4 PRs in flight it is getting difficult to origanize comments!